tensorflow_addons/optimizers/tests/cocob_test.py

```diff
@@ -0,0 +1,61 @@
+# Copyright 2020 The TensorFlow Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+"""
+Tests for COntinuos COin Betting (COCOB) Backprop optimizer
+"""
+
+import numpy as np
+import tensorflow as tf
+from tensorflow_addons.optimizers import COCOB
+
+
+def run_dense_sample(iterations, expected, optimizer):
+    var_0 = tf.Variable([1.0, 2.0], dtype=tf.dtypes.float32)
+    var_1 = tf.Variable([3.0, 4.0], dtype=tf.dtypes.float32)
+
+    grad_0 = tf.constant([0.1, 0.2], dtype=tf.dtypes.float32)
+    grad_1 = tf.constant([0.03, 0.04], dtype=tf.dtypes.float32)
+
+    grads_and_vars = list(zip([grad_0, grad_1], [var_0, var_1]))
+
+    for _ in range(iterations):
+        optimizer.apply_gradients(grads_and_vars)
+
+    np.testing.assert_allclose(var_0.read_value(), expected[0], atol=2e-4)
+    np.testing.assert_allclose(var_1.read_value(), expected[1], atol=2e-4)
+
+
+def test_dense_sample_with_default_alpha():
+    run_dense_sample(
+        iterations=100,
+        expected=[[-4.183396e16, -4.183396e16], [-4.1833639e16, -4.1833858e16]],
```

Comment:
maybe decrease the number of iterations? You are trying to test very small number with very large tolerance. In this case, I don't think it can be well tested.

> If the following equation is element-wise True, then allclose returns True.
>> absolute(a - b) <= (atol + rtol * absolute(b))

https://numpy.org/doc/stable/reference/generated/numpy.allclose.html

Reply:
- Hi, Okay. I will create a new Feature Request for the same.

- About the sparse implementation, I am facing some difficulty so thought first push the Dense one then with the help of the community and implement the sparse one. It was simple with TF v1.x but I am facing some problem with TF v2.x.

- Yes, I thought of testing with both low iteration and high iterations. But what you said absolutely made sense. So what is your suggestion? Should I keep one high iteration at least to test the extremes?

Reply:
> About the sparse implementation, I am facing some difficulty so thought first push the Dense one then with the help of the community and implement the sparse one. It was simple with TF v1.x but I am facing some problem with TF v2.x.

No worries. I can help you with that.

> Yes, I thought of testing with both low iteration and high iterations. But what you said absolutely made sense. So what is your suggestion? Should I keep one high iteration at least to test the extremes?

Just lower the iterations, maybe 5 or 10 is enough, and let the tolerance default if possible.

Reply:
The Feature Request: https://github.com/tensorflow/addons/issues/2064

Reply:
> No worries. I can help you with that.
> 
> >

Great. That will be helpful.

Reply:
> Just lower the iterations, maybe 5 or 10 is enough, and let the tolerance default if possible.

Done and Committed.